music_assistant/server/providers/builtin/__init__.py

```diff
@@ -507,59 +524,76 @@ async def get_stream_details(self, item_id: str) -> StreamDetails:
+            tracks = await self.mass.music.albums.tracks(
+                random_album.item_id, random_album.provider
+            )
+            for idx, track in enumerate(tracks, 1):
+                track.position = idx
+                result.append(track)
+        return result
+
+    async def _get_builtin_playlist_random_artist(self) -> list[Track]:
+        result: list[Track] = []
+        for random_artist in await self.mass.music.artists.library_items(
+            limit=1, order_by="random_fast"
+        ):
+            # use the function specified in the queue controller as that
+            # already handles unwrapping an artist by user preference
+            tracks = await self.mass.music.artists.tracks(
+                random_artist.item_id, random_artist.provider
             )
-            for idx, item in enumerate(res, 1):
-                item.position = idx
-                result.append(item)
-            return result
-        if builtin_playlist_id == RANDOM_TRACKS:
-            res = await self.mass.music.tracks.library_items(limit=500, order_by="random_fast")
-            for idx, item in enumerate(res, 1):
-                item.position = idx
-                result.append(item)
-            return result
-        if builtin_playlist_id == RANDOM_ALBUM:
-            for random_album in await self.mass.music.albums.library_items(
-                limit=1, order_by="random_fast"
-            ):
-                # use the function specified in the queue controller as that
-                # already handles unwrapping an album by user preference
-                tracks = await self.mass.music.albums.tracks(
-                    random_album.item_id, random_album.provider
-                )
-                for idx, track in enumerate(tracks, 1):
-                    track.position = idx
-                    result.append(track)
-                return result
-        if builtin_playlist_id == RANDOM_ARTIST:
-            for random_artist in await self.mass.music.artists.library_items(
-                limit=1, order_by="random_fast"
-            ):
-                # use the function specified in the queue controller as that
-                # already handles unwrapping an artist by user preference
-                tracks = await self.mass.music.artists.tracks(
-                    random_artist.item_id, random_artist.provider
-                )
-                for idx, track in enumerate(tracks, 1):
-                    track.position = idx
-                    result.append(track)
-                return result
-        if builtin_playlist_id == RECENTLY_PLAYED:
-            tracks = await self.mass.music.recently_played(100, [MediaType.TRACK])
             for idx, track in enumerate(tracks, 1):
                 track.position = idx
                 result.append(track)
-            return result
         return result
 
+    async def _get_builtin_playlist_recently_played(self) -> list[Track]:
```

Comment:
None of this code actually changed, but mypy had a real hard time with reusing "result" in different branches of the if statement and getting slightly different signatures (i guess another case of lists == invariant == bad?).

Reply:
It looks much better now - nice cleanup/restructure